packages/mailchimp/test/gzip.test.js

```diff
@@ -0,0 +1,87 @@
+import { expect } from 'chai';
+import * as undici from 'undici'
+import {Readable } from 'node:stream';
+import tar from 'tar-stream';
+import fs from 'node:fs'
+import gunzip from 'gunzip-maybe';
+import fixtureData from './fixtures/data.json' assert { type: 'json' }
+
+describe('gzip', () => {
+  it('should extract a tarball as json', (cb) => {
+    const extract = tar.extract();
+    let data = '';
+
+    extract.on('entry', function(header, stream, next) {
+      // header.name // filename
+      stream.on('data', function(chunk) {
+        data += chunk
+      });
+      
+      stream.on('end', function() {
+        next();
+      });
+
+      stream.resume();
+    });
+
+    extract.on('finish', function() {
+      const json = JSON.parse(data);
+      expect(json).eql(fixtureData)
+      cb()
+    });
+    
+    
+    fs.createReadStream('./test/fixtures/data.tar').pipe(extract)
+
+  })
+
+  // this is the same as above, but with nicer syntax
+  it('should extract a tarball as json with async', async () => {
+    const extract = tar.extract();
+    let data = '';
+
+    fs.createReadStream('./test/fixtures/data.tar').pipe(extract)
+
+    for await (const entry of extract) {
+      console.log(entry.header.name) // filename
+      for await (const d of entry) {
+        data = d;
+      }
+      entry.resume();
+    }
+
+    const json = JSON.parse(data);
+    expect(json).eql(fixtureData)
+  });
+
+
+  // To get this test to pass, you have to run `npx serve` from the fixtures folder
+  it('should download and extract a tarball with undici fetch', async () => {
+    const response = await undici.fetch('http://localhost:3000/data.tgz')
+
+    // note that we need to extract AND ungzip
```

Comment:
This is REALLY important, and where we were going wrong on Friday.

Undici can gzippdata on the wire for us, no problem. But it can't unzip and untar the actual response body - we have to do that ourselves.  So this first unzips and then untars the response.